### ISSUE TITLE
Make find_unused_port a local function instead of relying on test.support

### ISSUE THREAD

**Issue by luks29** [n8henrie/fauxmo#50]
- **Operating system and version**:
- **Python version**: 3.6.4
- **Fauxmo version** (`fauxmo --version`): 0.4.5
- **Echo device type** (e.g. Echo, Echo Plus, Dot): Echo Plus 2nd Gen 
- **[Echo Firmware Version](https://github.com/n8henrie/fauxmo/blob/dev/README.md#troubleshooting--faq)**: 

## My Issue
I cannot get fauxmo running as a startup service. Everything works perfectly when I use it under my virtualenv:
`fauxmo -c config.json -v`

But when I try:
`python -m fauxmo.cli -c config.json -v`

I get:
```
(venv) pi@raspi2:~$ python -m fauxmo.cli -c config.json -v
34
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/local/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/home/pi/venv/lib/python3.6/site-packages/fauxmo/cli.py", line 10, in <module>
    from fauxmo.fauxmo import main
  File "/home/pi/venv/lib/python3.6/site-packages/fauxmo/fauxmo.py", line 15, in <module>
    from test.support import find_unused_port
ModuleNotFoundError: No module named 'test.support'; 'test' is not a package
```

If I try starting it without loading my venv it either takes the wrong python version(2.7.?) or it brings the same error as above. 

And the same happens when I start it as service.

**Comment by n8henrie** [comment]
Hmm, I can't reproduce.
```
$ python --version
Python 3.6.4
$ python -m fauxmo.cli -c config.json -vv
2018-02-05 17:56:53 fauxmo:37       INFO     Fauxmo v0.4.5
2018-02-05 17:56:54 fauxmo:123      INFO     Starting UDP server
```

How did you install Fauxmo?

Can you try deleting your venv, recreating, installing fauxmo, and report back?

**Comment by luks29** [comment]
I installed it using pip install. I assume it was my python3.6 installation. After installing python with pyenv it worked.

**Comment by resuarez** [comment]
It may be out of the topic, but I am getting the same error ModuleNotFoundError: No module named 'test.support'; 'test' is not a package.  I am installing Python3.6 in Alpine Linux.  Unfortunately,  the installation doesn't include the test module and I don't know how to install it.  As solution, I've created the test folder and I've copied the support.py from another installation and it is working.  I am wondering if you know how to install that missing module.  And I am wondering if you can add the support.py in your distribution.  It will be more general.  Any case, thank you for you time.

**Comment by n8henrie** [comment]
How did you install Python? Are you just using the Alpine linux default installation?

**Comment by resuarez** [comment]
Yes, just adding the python3 package.  I've tried with the python3-dev, but I didn't get the test module.

**Comment by n8henrie** [comment]
I can reproduce.

```console
$ sudo docker run -it --rm alpine /bin/ash
Unable to find image 'alpine:latest' locally
latest: Pulling from library/alpine
bdf0201b3a05: Pull complete
Digest: sha256:28ef97b8686a0b5399129e9b763d5b7e5ff03576aa5580d6f4182a49c5fe1913
Status: Downloaded newer image for alpine:latest
/ # apk add python3
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.9/community/x86_64/APKINDEX.tar.gz
(1/11) Installing libbz2 (1.0.6-r6)
(2/11) Installing expat (2.2.6-r0)
(3/11) Installing libffi (3.2.1-r6)
(4/11) Installing gdbm (1.13-r1)
(5/11) Installing xz-libs (5.2.4-r0)
(6/11) Installing ncurses-terminfo-base (6.1_p20190105-r0)
(7/11) Installing ncurses-terminfo (6.1_p20190105-r0)
(8/11) Installing ncurses-libs (6.1_p20190105-r0)
(9/11) Installing readline (7.0.003-r1)
(10/11) Installing sqlite-libs (3.26.0-r3)
(11/11) Installing python3 (3.6.8-r2)
Executing busybox-1.29.3-r10.trigger
OK: 68 MiB in 25 packages
/ # apk add python3-dev
(1/2) Installing pkgconf (1.6.0-r0)
(2/2) Installing python3-dev (3.6.8-r2)
Executing busybox-1.29.3-r10.trigger
OK: 104 MiB in 27 packages
/ # python3 --version
Python 3.6.8
/ # python3 -c 'from test.support import find_unused_port'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'test'
```

[Another reason to use a different approach for this function](https://docs.python.org/3/library/test.html#module-test.support) :

> Note
> test.support is not a public module. It is documented here to help Python developers write tests. The API of this module is subject to change without backwards compatibility concerns between releases.

**Comment by n8henrie** [comment]
Closed in #77

**Comment by resuarez** [comment]
Great thank you
